src/main/java/software/aws/neptune/opencypher/OpenCypherConnectionProperties.java

```diff
@@ -264,12 +242,16 @@ public void setConnectionPoolSize(final int connectionPoolSize) throws SQLExcept
     protected void validateProperties() throws SQLException {
         // If IAMSigV4 is specified, we need the region provided to us.
         if (getAuthScheme() != null && getAuthScheme().equals(AuthScheme.IAMSigV4)) {
-            final String region = System.getenv().get("SERVICE_REGION");
-            if (region == null) {
-                throw missingConnectionPropertyError(
-                        "A Region must be provided to use IAMSigV4 Authentication. Set the SERVICE_REGION environment variable to the appropriate region, such as 'us-east-1'.");
+            if ("".equals(getRegion())) {
```

Comment:
Migrate this logic to an abstract function in base and pull out of all specific functions.

Reply:
Abstracted the check into a protected function in parent class called by different query languages.